### PR TITLE
Feat: autocomplete searchbox

### PR DESCRIPTION
**Pull request by agtenr**
## Type of change

- [ ] Bugfix
- [ ] Component improvement
- [x] New component
- [ ] New package
- [ ] Other

## Description
This PR contains an autocomplete search box:

- Extends the fluent ui search box
- Adds an onResolveSuggestion method for rendering autocomplete options
- Re-uses the accessible controls from Connect
- Add arrow down and tab controls

# Improvements:
This component currently supports all features for the connect implementation. Extra funcionality is needed for Search:
 - On focus --> for personal search history
 - Custom Renderer for adding icons/controls to suggestions

### CODE REVIEW

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/AutocompleteSearchBox.types.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	 * The callback method to resolve the suggestions for the autocomplete search box.
+	 * @param searchValue The search value to resolve the suggestions for.
+	 */
+	onResolveSuggestions: (searchValue: string) => Promise<string[]>;
+
+	/**
+	 * The debounce time in milliseconds for the search box to start resolving the suggestions.
+	 * @default 0
+	 */
+	debounceTime?: number;
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/AutocompleteSearchBox.types.ts:29]: perhaps change this to `resolveDelay` to keep it consistent with https://developer.microsoft.com/en-us/fluentui#/controls/web/pickers#IBasePickerProps

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/AutocompleteSearchBox.types.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	 * Optional flag to render the accessible confirmation button.
+	 * @default true
+	 */
+	withAccessibleConfirmationButton?: boolean;
+
+	/**
+	 * The aria description for the confirmation button.
+	 */
+	confirmationButtonAriaDescription?: string;
+}
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/AutocompleteSearchBox.types.ts:52]: i would also add `calloutProps` to be able to override the callout when necessary

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/HighlightedSuggestion/HighlightedSuggestion.base.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	// Return the suggestion if no filter is provided
+	if (!!!filter) return <>{props.text}</>;
+
+	try {
+		// Split text on highlight term, include term itself into parts, ignore case
+		const parts = React.useMemo(() => text.split(new RegExp(`^(${filter})`, "gi")), [text, filter]);
+
+		return (
+			<span>
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/HighlightedSuggestion/HighlightedSuggestion.base.tsx:22]: why do we need the start/end spaces?

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/HighlightedSuggestion/HighlightedSuggestion.base.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+						? classNames.rootHighlighted
+						: classNames.root;
+
+					return (
+						<span key={i} className={className}>
+							{part}
+						</span>
+					);
+				})}{" "}
+			</span>
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/HighlightedSuggestion/HighlightedSuggestion.base.tsx:35]: `span` containing `span`s seems odd. Can this html be simplified?

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { ArgsTable, Meta, Story, Canvas } from "@storybook/addon-docs";
+import { AutocompleteSearchBox } from "../AutocompleteSearchBox";
+import * as stories from "./AutocompleteSearchBox.stories";
+
+<Meta title="Components/AutocompleteSearchBox" component={AutocompleteSearchBox} />
+
+# AutocompleteSearchBox
+
+The autocomplete searchbox is a component that expands on the Fluent UI React Searchox with an autocomplete dropdown.
+The component is inspired and build upon the Microsoft example: [https://github.com/microsoft/autocomplete-search-box]()
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx:10]:
```suggestion
The component is inspired and built upon the Microsoft example: [https://github.com/microsoft/autocomplete-search-box]()
```

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+When typing the searchbox, autocomplete suggestions are dynamically fetched and displayed in a dropdown below the searchbox.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx:11]:
```suggestion
When typing in the searchbox, autocomplete suggestions are dynamically fetched and displayed in a dropdown below the searchbox.
```

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+### Basic usage
+
+This example shows the basic usage of the AutocompleteSearchBox component.
+The only required property is an `onResolveSuggestions` funtion that return a promise with the suggestions.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx:16]:
```suggestion
The only required property is an `onResolveSuggestions` function that returns a promise with the suggestions.
```

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+<Canvas withToolbar={true}>
+	<Story story={stories.Basic} name="Basic usage" />
+</Canvas>
+
+### Advance usage
+
+This example shows the advance usage of the AutocompleteSearchBox component.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx:24]:
```suggestion
This example shows the advanced usage of the AutocompleteSearchBox component.
```

File: packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+It is possible to control the value outside of the component by using the `value` and `onChange` properties.
+It also shows some examples of the debounce functionality and how to use the `onSearch` event.
+There is also an optoin to disable the suggestion highlighting.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/AutocompleteSearchBox/stories/AutocompleteSearchBox.stories.mdx:27]:
```suggestion
There is also an option to disable the suggestion highlighting.
```

**Review by NickSevens — CHANGES_REQUESTED** [review]

**Comment by agtenr** [comment]
@NickSevens Thanks for the review, I updated the code:
 - Fixed typos
 - Renamed debounce property
 - Exposed calloutProps
 - Removed unnecessary spans/spaces in highlighted component

**Review by NickSevens — COMMENTED** [review]
the `calloutProps` could probably be a bit more dynamic still (e.g., how would I now decide on the beak being visible?), but let's go ahead with the PR for now. :) This can be improved in the future if the need is there.

**Review by NickSevens — APPROVED** [review]

**Pull request merged**